### ISSUE TITLE
How do I not get banned from landroid cloud service?

### ISSUE THREAD

**Issue by shahmirthesquid** [MTrab/landroid_cloud#421]
### Describe the issue

Hi, I've been enjoying this integration very much. But I regularly get banned from the landroid cloud service and then am unable to make any commands whatsoever. I always have to unpair my landroid then repair it. Which gives me about a 2-5min window to make commands before it bans me again 😢  Also I believe that getting banned from landroid cloud stops my landroid from connecting to wifi. As the wifi symbol only shows during that 2-5min unban window I mentioned previously.

This has been very frustrating to me. I have no way to remotely stop my landroid if it gets into trouble. Or to send it home if I'm going out. Am I doing something wrong? Does anyone have any tips for not getting banned?

Many users seem to be using this integration without an issue. Is it possible I am getting banned because I use both the HomeAssistant and Landroid apps interchangeably? I have the GPS tracker addon for my landroid so I regularly open the Landroid app to see it's location.

I have read in other posts that making a brand new account will give the command functionality back. But I don't want to do that until I understand the reason I am getting banned, and how to prevent getting banned in the future. Please any help would be very appreciated ❤️ 

### What version of Home Assistant Core has the issue?

core-2023.8.3

### What version of the Landroid Cloud integration do you have installed

v3.0.5

### What type of installation are you running?

Home Assistant OS

### Which make and model is the mower used for this integration?

Worx WR140

### Diagnostics information (NOT log entries!)

[config_entry-landroid_cloud-974ec2a3acc0802901727d0303356334.json.txt](https://github.com/MTrab/landroid_cloud/files/12468988/config_entry-landroid_cloud-974ec2a3acc0802901727d0303356334.json.txt)

### Relevant log entries

```txt
None
```

**Comment by silkyclouds** [comment]
yeah same here, this integration is working, then not working, working, not working...I'm back to the official landroid app for now. This card is not brining a lot of advantages anyway.

**Comment by MTrab** [comment]
No idea why you get banned, unless you make excessive calls to the API.
Are you 100% sure you get banned?

**Comment by nrm94ee330** [comment]
I also started having this issue.. it had been working fine for almost a year or so without issues.. and suddenly a few days back, it says "failed to setup".. I dont want to unpair and repair my landroid.. then I will have to redo all the setting in the lawn perimiter, zone etc.. that is quite a hassle !

**Comment by shahmirthesquid** [comment]
> I also started having this issue.. it had been working fine for almost a year or so without issues.. and suddenly a few days back, it says "failed to setup".. I dont want to unpair and repair my landroid.. then I will have to redo all the setting in the lawn perimiter, zone etc.. that is quite a hassle !

I think your issue might be different from mine. I also started getting the "failed to setup" message yesterday. But it went away after I updated to v3.06 of landroid cloud which was just released. You can update yours by going into the HACS tab in home assistant. It should tell you that some integrations have updates available.

In my original post, the integration was able to setup just fine and even read data from the mower. But I was unable to issues commands like "start mowing" or "go home" in both home assistant and the landroid app. My understanding is that when you get banned from the landroid service your account ID is simply prohibited from sending commands.

**Comment by shahmirthesquid** [comment]
> No idea why you get banned, unless you make excessive calls to the API. Are you 100% sure you get banned?

I did setup an automation to reload the landroid cloud service every hour. I did this because I was suffering from the same bug as this person:

https://github.com/MTrab/landroid_cloud/issues/385

I disabled the automation after a day. I am not sure if this is the reason I got banned, but it is definitely possible.

 I am 99% sure I got banned, as my symptoms seem to match every other post describing the effects of getting banned. 

It is raining here so I haven't been able to test again. I will post back here in a day or two with an update about whether the problem resolved itself.

**Comment by nrm94ee330** [comment]
@MTrab I just saw that you posted an update.. I installed it.. and it works now..
Thanks for such an amazingly fast turnaround !

**Comment by pascalprey** [comment]
> I also started having this issue.. it had been working fine for almost a year or so without issues.. and suddenly a few days back, it says "failed to setup".. I dont want to unpair and repair my landroid.. then I will have to redo all the setting in the lawn perimiter, zone etc.. that is quite a hassle ! looking at the logs after trying to resetup.. got
> 
> Traceback (most recent call last): File "/usr/src/homeassistant/homeassistant/config_entries.py", line 388, in async_setup result = await component.async_setup_entry(hass, self) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/landroid_cloud/**init**.py", line 63, in async_setup_entry result = await _async_setup(hass, entry) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/config/custom_components/landroid_cloud/**init**.py", line 206, in _async_setup await hass.async_add_executor_job(cloud.connect) File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run result = self.fn(*self.args, **self.kwargs) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/pyworxcloud/**init**.py", line 259, in connect self._fetch() File "/usr/local/lib/python3.11/site-packages/pyworxcloud/**init**.py", line 611, in _fetch self._mowers = self._api.get_mowers() ^^^^^^^^^^^^^^^^^^^^^^ File "/usr/local/lib/python3.11/site-packages/pyworxcloud/api.py", line 119, in get_mowers mower["firmware_version"] = "{:.2f}".format(mower["firmware_version"]) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ValueError: Unknown format code 'f' for object of type 'str'
> 
> Could it be possible that the firmware version shows "Version 3.32.0+1" and the "+" sign is throwing it off ?

Same problem here:

`Error setting up entry Worx - abc@def.gh for landroid_cloud
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 388, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/landroid_cloud/__init__.py", line 63, in async_setup_entry
    result = await _async_setup(hass, entry)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/landroid_cloud/__init__.py", line 206, in _async_setup
    await hass.async_add_executor_job(cloud.connect)
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyworxcloud/__init__.py", line 259, in connect
    self._fetch()
  File "/usr/local/lib/python3.11/site-packages/pyworxcloud/__init__.py", line 611, in _fetch
    self._mowers = self._api.get_mowers()
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyworxcloud/api.py", line 119, in get_mowers
    mower["firmware_version"] = "{:.2f}".format(mower["firmware_version"])
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Unknown format code 'f' for object of type 'str'
`

**Comment by MTrab** [comment]
Update to latest version

**Comment by shahmirthesquid** [comment]
Hi I just wanted to give a quick update. I had re-paired my landroid multiple times as stated above, but during all those times I reused the same name for my landroid which was "grunu". When I re-paired and set a new name for my landroid as "gruunu" with an extra "U" all of my symptoms of being banned from the landroid service went away. I did not have to make a new account, I just needed to change my landroid's name. Which seems to be enough for the worx server to think it was a different landroid. Now I can send commands through both the landroid app, and through the home assistant integration with no problems. I think it would be useful for other users to add changing the name of your landroid to the "banned from landroid service" troubleshooting instructions. Thank you again to the developer for making this incredibly useful integration ❤️
